taskRequests/details/script.js

```diff
@@ -251,4 +267,162 @@ const renderTaskRequest = async () => {
   }
 };
 
+function showToast(message, type) {
+  toast.innerHTML = `<div class="message">${message}</div>`;
+  toast.classList.remove('hidden');
+
+  if (type === 'success') {
+    toast.classList.add('success');
+    toast.classList.remove('failure');
+  } else if (type === 'failure') {
+    toast.classList.add('failure');
+    toast.classList.remove('success');
+  }
+
+  setTimeout(() => {
+    toast.classList.add('hidden');
+    toast.innerHTML = '';
+  }, 5000);
+}
+
+async function getAssigneeName() {
+  let userName = '';
+  let res;
+  if (taskRequest.approvedTo) {
+    try {
+      res = await fetch(
+        `${API_BASE_URL}/users/userId/${taskRequest.approvedTo}`,
+      );
+    } catch (error) {
+      console.error(error);
+    }
+    if (res.ok) {
+      const userData = await res.json();
+      userName = userData.user.first_name;
+    }
```

Comment:
this if block can be written in the try block, then you wouldn't have to put `res` outside the try block

Reply:
Sorry I didn't get it the entire if block is within try catch block.

```js
async function approveTaskRequest(userId) {
  try {
    const res = await fetch(`${API_BASE_URL}/taskRequests/approve`, {
      credentials: 'include',
      method: 'PATCH',
      body: JSON.stringify({
        taskRequestId,
        userId,
      }),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (res.ok) {
      showToast('Task Approved Successfully', 'success');
      taskRequest = await fetchTaskRequest();
      requestorsContainer.innerHTML = '';
      renderRequestors(taskRequest?.requestors);
    } else {
      showToast(errorMessage, 'failure');
    }
  } catch (e) {
    showToast(errorMessage, 'failure');
    console.error(e);
  }
}
```